### PR TITLE
updates edit program link to redirect to last block

### LINKED ISSUE #3887: Error when editing a program
**Describe the bug**
As a CiviForm Admin, I'd like to edit a program. When I select "Edit" on a specific program, I see an error page with this text: "_services.program.ProgramBlockDefinitionNotFoundException: Block not found in Program (ID 1079) for block definition ID 1_" and this URL: **https://civiform.seattle.gov/admin/programs/1079/blocks/1/edit**.

I have a work-around: I just remove a section of the URL: **/blocks/1**

**To Reproduce**

1. Login as a CiviForm Admini
2. Navigate to the Program Dashboard
3. Select "Edit" next to one program

**Expected behavior**
Would expect to be routed to the "Edit" page for the program.

**Screenshots**
image

Attached

Error in program edit

### PR DESCRIPTION
**Pull request by seanmalbert**
### Description

This updates the edit program link to go to `/blocks` which gets the last block in a program and redirects to `/edit` with the last block ID.

## Release notes

Updates the edit program link to go to `/blocks` which gets the last block in a program and redirects to `/edit` with the last block ID.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/application.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #3887

### CODE REVIEW

File: browser-test/src/admin_program_creation.test.ts [hunk 1]
```diff
@@ -181,6 +181,18 @@ describe('program creation', () => {
     ])
   })
 
+  it('delete first block and edit', async () => {
```
Comment (shanemc-goog) [inline browser-test/src/admin_program_creation.test.ts:184]: Can you add a comment why we're focusing on "delete first block". ie Why should someone later on not make it the 3rd block?

**Review by shanemc-goog — APPROVED** [review]

File: browser-test/src/admin_program_creation.test.ts (hunk 1 above)
Comment (shanemc-goog) [inline browser-test/src/admin_program_creation.test.ts:184]: Might be worth adding a test to delete the last block too :D

**Review by shanemc-goog — APPROVED** [review]

**Review by bion — APPROVED** [review]

File: browser-test/src/admin_program_creation.test.ts (hunk 1 above)
Comment (seanmalbert) [reply browser-test/src/admin_program_creation.test.ts:189]: added a comment clarifying why we're specifically testing deleting the first block

Comment (seanmalbert) [reply browser-test/src/admin_program_creation.test.ts:189]: Added test that deletes the last block

**Pull request merged**